File: backend-java/src/main/java/com/bffgin/backend/auth/Claims.java
package com.bffgin.backend.auth;

/**
 * user_id解決(UserResolver)に必要な最小限のクレームに加え、外部公開API向けの`azp`
 * (authorized party、Client Credentials Grantのクライアントid)も保持する。
 * 内部REST/gRPCのuser_id解決にはazpは使わないため、Phase 1では省略していた
 */
public record Claims(String sub, String iss, String azp) {
}

File: backend-kotlin/src/main/kotlin/com/bffgin/backend/auth/Claims.kt
package com.bffgin.backend.auth

/**
 * user_id解決(UserResolver)にはsub/issのみ必要だが、外部公開API
 * (RequireExternalClientAuth相当、ExternalAuth.kt参照)はazp(authorized party)クレームで
 * Client Credentials Grantのクライアントを識別する必要があるため保持する。
 * azpクレームが無いトークン(ローカルHMAC/RSA発行分は通常持たない)は空文字列になる
 */
data class Claims(val sub: String, val iss: String, val azp: String = "")
